galley/pkg/kube/converter/converter.go

```diff
@@ -206,6 +216,39 @@ func kubeIngressResource(cfg *Config, _ resource.Info, name resource.FullName, _
 	return []Entry{e}, nil
 }
 
+func kubeServiceResource(_ *Config, _ resource.Info, name resource.FullName, _ string, u *unstructured.Unstructured) ([]Entry, error) {
+	var service corev1.Service
+	if err := convertJSON(u, &service); err != nil {
+		return nil, err
+	}
+	se := networking.ServiceEntry{
+		// TODO: What's the difference between the name argument and <service.Name>/<service.Namespace>?
+		Hosts:     []string{fmt.Sprintf("%s.%s.svc.cluster.local", service.Name, service.Namespace)},
+		Addresses: append([]string{service.Spec.ClusterIP}, service.Spec.ExternalIPs...),
+	}
+	for _, kubePort := range service.Spec.Ports {
+		istioPort := &networking.Port{
+			Name:   kubePort.Name,
+			Number: uint32(kubePort.Port),
+		}
+		// TODO: Is there existing code that handles magic port prefixes?
+		if kubePort.Name == "http" || strings.HasPrefix(kubePort.Name, "http-") {
+			if kubePort.Protocol != "TCP" {
+				return nil, fmt.Errorf("port %s: HTTP over %s not supported", kubePort.Name, kubePort.Protocol)
+			}
+			istioPort.Protocol = "HTTP"
+		} else {
+			istioPort.Protocol = string(kubePort.Protocol)
+		}
+		se.Ports = append(se.Ports, istioPort)
+	}
+	return []Entry{{
+		Key:          name,
+		CreationTime: service.CreationTimestamp.Time,
+		Resource:     &se,
+	}}, nil
+}
+
```

Comment:
just FYI, this is very insufficient and wont work even in the basic case.
You need to set the resolution to STATIC.
Use the model.ParseProtocol and populate the port protocol
Do not hardcode the svc.cluster.local as the domain is configurable. So you might have to obtain this from some global flag.

Reply:
Pilot uses DomainSuffix for the domain, but I do not see an equivalent in Galley, is there one? @ozevren

Reply:
I've added one for the Ingress work:
https://github.com/istio/istio/blob/913b459130045fc4846a36c46c05a48da88776bb/galley/pkg/runtime/config.go#L25
It should be accessible in the projection code, but not in the front-end converter.

Reply:
- Reusing -domain flag
- resolution set
- using kube.ConvertPort now